cli/module_graph2.rs

```diff
@@ -336,14 +359,30 @@ impl Module {
         col: desc.col,
         line: desc.line,
       };
-      let specifier =
-        self.resolve_import(&desc.specifier, Some(location.clone()))?;
+
+      // In situations where there is a potential issue with resolving the
+      // import specifier, that ends up being a module resolution error for a
+      // code dependency, we should not throw in the `ModuleGraph` but instead
+      // wait until runtime and throw there, as with dynamic imports they need
+      // to be catchable, which means they need to be resolved at runtime.
+      let maybe_specifier =
+        match self.resolve_import(&desc.specifier, Some(location.clone())) {
+          Ok(specifier) => Some(specifier),
+          Err(any_error) => {
+            match any_error.downcast_ref::<ModuleResolutionError>() {
+              Some(ModuleResolutionError::ImportPrefixMissing(_, _)) => None,
+              _ => {
+                return Err(any_error);
+              }
+            }
+          }
+        };
```

Comment:
I think this bit won't return error for either dynamic or static imports - is that desired? Maybe error could be ignored only for dynamic imports?

Reply:
It is desired because if I don't, then transient dynamic imports become non-catchable. (Part of error_014 test cases).

Reply:
Let me rephrase it: will error be returned for errors in static imports?

Reply:
Yes, but not by the graph or prepare module, but instead the load module.  There is an integration test that tests for it, and it passes.